build:created database schema for users

epun21403 opened the pull request:
# Pull Request Template

## Description

Description: 
Added Prisma as a dependency and created the schema for users. 
Added "DATABASE_URL" to env file in the following format example:
postgresql://USER:PASSWORD@HOST:PORT/DATABASE?schema=SCHEMA
Here's a short explanation of each component:
-USER: The name of your database user
-PASSWORD: The password for your database user
-HOST: The name of your host name (for the local environment, it is localhost)
-PORT: The port where your database server is running (typically 5432 for PostgreSQL)
-DATABASE: The name of the [database](https://www.postgresql.org/docs/12/manage-ag-overview.html)
-SCHEMA: The name of the [schema](https://www.postgresql.org/docs/12/ddl-schemas.html) inside the database

I ran the following line in the terminal to create the tables and database:
npx prisma migrate dev --name init

## Type of Change

Please label your pull request as appropriate for the type of change it makes and remove labels that are not relevant below:

- build: affects the application build process or repository

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.md) for this project.
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] If applicable, I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Reviewers

@afrodynamic 

## Additional Notes

Please add any additional notes or context about this pull request here.

afrodynamic reviewed (changes requested):
Make sure to either delete the `migrations` folder or add it to `.gitignore`, these files don't need to be included in the repo for our usage. The rest LGTM, good work!

```
prisma/migrations/
prisma/migrations/migration_lock.toml
prisma/migrations/20230612195154_init/
prisma/migrations/20230612195154_init/migration.sql
```

afrodynamic reviewed (approved):
LGTM

epun21403 commented:
6edd5c0
Added controllers and API routes for users table and favorites table
Added POST for users tables
Added POST, GET, and DELETE for favorites table

afrodynamic commented on server/controllers/favoritesController.ts:
```diff
@@ -0,0 +1,53 @@
+};
+
+export const removePlantFromFavorite = async (request: Request, response: Response) => {
+  const { plantId } = request.params;
```
I think we should keep the request format consistent here, the other controller functions are using only using `request.body` so I'd change it here as such:
 
```suggestion
  if (!request.body || !request.body.plantId || !request.body.userId) {
    response.status(500).send("Not all information included");
    return;
  }
```
I'll add an additional suggestion where you reference plantId that'd need to be changed too

afrodynamic commented on server/controllers/favoritesController.ts:
```diff
@@ -0,0 +1,53 @@
+  try {
+    const deletePlant = await prisma.favorites.deleteMany({
+      where: {
+        plantId: Number(plantId),
```
As noted above:

```suggestion
        plantId: Number(request.body.plantId),
        
```

afrodynamic commented on server/routes/favoritesRoutes.ts:
```diff
@@ -0,0 +1,11 @@
+
+router.get('/', getPlantsFromFavorite);
+router.post('/', addPlantToFavorite);
+router.delete('/:plantId', removePlantFromFavorite);
```
For the consistency changes you would want to remove the variable param here too:

```suggestion
router.delete('/', removePlantFromFavorite);
```

afrodynamic reviewed (changes requested):
Left some suggestions to maintain consistency, the rest LGTM

epun21403 commented:
> Left some suggestions to maintain consistency, the rest LGTM

Implemented recommended changes to maintain consistency.

afrodynamic reviewed (approved):
LGTM

jnovak79 commented on client/src/features/favoritePlants/FavoritePlants.tsx:
Line 12, remove console.log

Line 16, remove err from console.log

jnovak79 commented on client/src/features/plantDetails/PlantDetails.tsx:
Line 26, remove console.log

line 29 remove error from console.log

jnovak79 reviewed (changes requested):
Some feedback to remove console.logs. Otherwise LGTM

jnovak79 reviewed (approved):
LGTM

afrodynamic reviewed (approved):
LGTM, submitting a commit to update some of the console.logs and return statement logic
